client/src/components/Tour/runTour.js

```diff
@@ -27,22 +27,34 @@ function getElement(selector) {
 }
 
 // wait for element
-function waitForElement(selector, resolve, reject, tries) {
-    if (selector) {
-        const el = getElement(selector);
-        const rect = el?.getBoundingClientRect();
-        const isVisible = !!(rect && rect.width > 0 && rect.height > 0);
-        if (el && isVisible) {
-            resolve();
-        } else if (tries > 0) {
-            setTimeout(() => {
-                waitForElement(selector, resolve, reject, tries - 1);
-            }, delay);
+function waitForElement(selector, resolve, reject, tries, prerequisite) {
+    try {
+        if (selector) {
+            const el = getElement(selector);
+            const rect = el?.getBoundingClientRect();
+            const isVisible = !!(rect && rect.width > 0 && rect.height > 0);
+            if (el && isVisible) {
+                resolve();
+            } else if (tries > 0) {
+                // First, try to click the prerequisite element if it exists
+                if (prerequisite) {
```

Comment:
Does this not cause multiple clicks on the `prerequisite` element i.e. keeps opening and closing the tool panel, maybe a conditional step would work better. Thank you for working on this.

Reply:
Ah thats a good point 👉 
Will fix that

Reply:
Not doing prerequisites anymore as mentioned below